pkg/machinery/config/provider.go

```diff
@@ -20,15 +20,25 @@ import (
 
 // Provider defines the configuration consumption interface.
 type Provider interface {
+	// Config parts accessor.
 	Version() string
 	Debug() bool
 	Persist() bool
 	Machine() MachineConfig
 	Cluster() ClusterConfig
+
 	// Validate checks configuration and returns warnings and fatal errors (as multierror).
 	Validate(RuntimeMode, ...ValidationOption) ([]string, error)
-	String(encoderOptions ...encoder.Option) (string, error)
-	Bytes(encoderOptions ...encoder.Option) ([]byte, error)
+
+	// Bytes returns source YAML representation (if available) or does default encoding.
+	Bytes() ([]byte, error)
+
+	// Encode configuration to YAML using the provided options.
+	EncodeString(encoderOptions ...encoder.Option) (string, error)
```

Comment:
I renamed `Bytes` and `Strings` to `EncodeXXX`, as with custom options they should always do marshaling to YAML (as options affect the way config is marshaled).

The method `Bytes` returns original `[]byte` representation if available.